cmd/clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -273,6 +287,32 @@ func (c *testClusterClient) DeleteClusters(ns string) error {
 	return nil
 }
 
+func (c *testClusterClient) DeleteUnstructuredObjects(ns string, u *unstructured.Unstructured) error {
```

Comment:
I can remove it, I thought to use it for cleaning up all the external objects, but it might not be necessary

Reply:
The way I read the logic, it's doing this:

1. Loop through all objects by namespace
    1. If the requested namespace matches the loop namespace, skip it (?????)
    1. Otherwise, loop through all objects
        1. If the loop object's kind is different than the requested object, remove it from the list
1. Check the requested namespace
    1. If the requested namespace is empty, wipe out the internal storage for clusters
    1. Otherwise, delete all clusters for the requested namespace from internal storage

I'm confused by `1.i` and the last bit about clusters.